gradle/lib/dependabot/gradle/update_checker/version_finder.rb

```diff
@@ -76,24 +79,39 @@ def versions
         attr_reader :dependency, :dependency_files, :credentials,
                     :ignored_versions, :forbidden_urls, :security_advisories
 
+        sig { params(possible_versions: T::Array[T.untyped]).returns(T::Array[T.untyped]) }
         def filter_prereleases(possible_versions)
           return possible_versions if wants_prerelease?
 
-          possible_versions.reject { |v| v.fetch(:version).prerelease? }
+          filtered = possible_versions.reject { |v| v.fetch(:version).prerelease? }
+          if possible_versions.count > filtered.count
+            Dependabot.logger.info("Filtered out #{possible_versions.count - filtered.count} pre-release versions")
+          end
+          filtered
         end
 
+        sig { params(possible_versions: T::Array[T.untyped]).returns(T::Array[T.untyped]) }
         def filter_date_based_versions(possible_versions)
           return possible_versions if wants_date_based_version?
 
-          possible_versions
-            .reject { |v| v.fetch(:version) > version_class.new(1900) }
+          filtered = possible_versions.reject { |v| v.fetch(:version) > version_class.new(1900) }
+          if possible_versions.count > filtered.count
+            Dependabot.logger.info("Filtered out #{possible_versions.count - filtered.count} date-based versions")
+          end
+          filtered
         end
 
+        sig { params(possible_versions: T::Array[T.untyped]).returns(T::Array[T.untyped]) }
         def filter_version_types(possible_versions)
-          possible_versions
-            .select { |v| matches_dependency_version_type?(v.fetch(:version)) }
+          filtered = possible_versions.select { |v| matches_dependency_version_type?(v.fetch(:version)) }
+          if possible_versions.count > filtered.count
+            diff = possible_versions.count - filtered.count
+            Dependabot.logger.info("Filtered out #{diff} versions not matching the type")
```

Comment:
not matching the type? Can that be explained some more?

Reply:
Just looked up what these suffixes are actually called, they're artifact classifiers, so I updated the log message. Thanks for catching that.